doc/build/kconfig/setting.rst

```diff
@@ -243,25 +243,42 @@ Note that conditions from surrounding top-level ``if``\ s are propagated to
 symbol properties, so the above ``default`` is equivalent to
 ``default 32 if BOARD_MY_BOARD``.
 
-.. warning::
+Multiple symbol definitions
+---------------------------
+
+When defining a symbol in multiple locations, dependencies are ORed together
+rather than ANDed together. It is not possible to make the dependencies of a
+symbol more restrictive by defining it in multiple locations.
+
+For example, the direct dependencies of the symbol below becomes
+``DEP1 || DEP2``:
+
+.. code-block:: none
+
+   config FOO
+      ...
+      depends on DEP1
 
-   When defining a symbol in multiple locations, dependencies are ORed together
-   rather than ANDed together. It is not possible to make the dependencies of a
-   symbol more restrictive by defining it in multiple locations.
+   config FOO
+      ...
+      depends on DEP2
 
-   For example, the direct dependencies of the symbol below becomes
-   ``DEP1 || DEP2``:
+.. warning::
+   Symbols without explicit dependencies are treated as being dependent on the
+   ``y`` symbol. The definition below will result in ``FOO`` always being
```

Comment:
I believe it was already discussed here that this description is imprecise as it is not dependent on `y`.

Instead the way `depends on <symbol>` work is clearly defined by the Kconfig language.
https://github.com/zephyrproject-rtos/zephyr/pull/51110#discussion_r994260924

Reply:
Good point, I have reworded this section to avoid the `DEP1 || DEP2` example, which makes it awkward to explain the behaviour of a promptless symbol without resorting to `y`.